ansible_base/jwt_consumer/common/auth.py

```diff
@@ -146,27 +166,122 @@ def validate_token(self, token, decryption_key):
+            logger.error("Unable to process rbac permissions because user or token is not defined, please call authenticate first")
+            return
+
+        for system_role_name in self.token.get("global_roles", []):
+            logger.debug(f"Processing system role {system_role_name} for {self.user.username}")
+            try:
+                rd = RoleDefinition.objects.get(name=system_role_name)
+                rd.give_global_permission(self.user)
+                logger.info(f"Granted user {self.user.username} global role {system_role_name}")
+            except RoleDefinition.DoesNotExist:
+                logger.error(f"Unable to grant {self.user.username} system level role {system_role_name} because it does not exist")
+                continue
+
+        for object_role_name in self.token.get('object_roles', {}).keys():
+            rd = None
+            try:
+                rd = RoleDefinition.objects.get(name=object_role_name)
+            except RoleDefinition.DoesNotExist:
+                logger.error(f"Unable to grant {self.user.username} object role {object_role_name} because it does not exist")
+                continue
+
+            object_type = self.token['object_roles'][object_role_name]['content_type']
+            object_indexes = self.token['object_roles'][object_role_name]['objects']
+
+            for index in object_indexes:
+                object_data = self.token['objects'][object_type][index]
+                resource, obj = self.get_or_create_resource(object_type, object_data)
+                if resource is not None:
+                    rd.give_permission(self.user, obj)
+                    logger.info(f"Granted user {self.user.username} role {object_role_name} to object {obj.name} with ansible_id {object_data['ansible_id']}")
+
+    def get_or_create_resource(self, content_type: str, data: dict) -> Tuple[Optional[Resource], Optional[Model]]:
+        """
+        Gets or creates a resource from a content type and its default data
+
+        This can only build or get organizations or teams
+        """
+        object_ansible_id = data['ansible_id']
+        try:
+            resource = Resource.objects.get(ansible_id=object_ansible_id)
+            logger.debug(f"Resource {object_ansible_id} already exists")
+            return resource, resource.content_object
+        except Resource.DoesNotExist:
+            pass
+
+        # The resource was missing so we need to create its stub
+        if content_type == 'team':
+            # For a team we first have to make sure the org is there
+            org_id = data['org']
+            organization_data = self.token['objects'][self.org_content_type.model][org_id]
+
+            # Now that we have the org we can build a team
+            resource, org = self.get_or_create_resource(self.org_content_type.model, organization_data)
+            team, created = self.team_content_type.model_class().objects.get_or_create(
```

Comment:
Instead of calling `self.team_content_type.model_class().objects.get_or_create`, would it be cleaner to call `resource.create_resource`?

https://github.com/ansible/django-ansible-base/blob/9f0f05168dbcb18c1f542aec480b80eedb5ffc3e/ansible_base/resource_registry/models/resource.py#L99

Reply:
Will follow up in a separate PR.

Reply:
Actually that's a classmethod, and doesn't do quite what i expected. Maybe don't bother.